gapis/trace/android/validate/validate.go

```diff
@@ -117,26 +98,54 @@ func CheckEqualTo(num float64) Checker {
 	}
 }
 
-// CheckApproximateTo returns a checker that checks that values are within a margin of the given value.
-func CheckApproximateTo(num, err float64) Checker {
+// Average returns a checker that checks whether the average values meets the condition
+func Average(check func (float64) bool) Checker {
 	return func(column *perfetto_service.QueryResult_ColumnValues, columnType perfetto_service.QueryResult_ColumnDesc_Type) bool {
 		longValues := column.GetLongValues()
 		doubleValues := column.GetDoubleValues()
+		total := float64(0.0)
 		for i := 0; i < sampleCounter; i++ {
 			if columnType == perfetto_service.QueryResult_ColumnDesc_LONG {
-				if math.Abs(num-float64(longValues[i])) > err {
-					return false
-				}
+				total += float64(longValues[i])
 			} else if columnType == perfetto_service.QueryResult_ColumnDesc_DOUBLE {
-				if math.Abs(num-doubleValues[i]) > err {
-					return false
-				}
+				total += doubleValues[i]
 			}
 		}
-		return true
+		return check(total / sampleCounter)
 	}
 }
 
+// CheckLargerThanZero is a checker that checks that the values are all greater than zero.
+func CheckLargerThanZero() Checker {
+	return ForeachValue(func(v float64) bool {
+		if v <= 0.0 {
+			return false
+		}
+		return true
+	})
+}
+
+// CheckEqualTo returns a checker that checks that all returned value equal the given value.
+func CheckEqualTo(num float64) Checker {
+	return ForeachValue(func (v float64) bool {
+		if v != num {
+			return false
+		}
+		return true
+	})
+}
+
+// CheckAverageApproximateTo checks whether the average of the values is within
+// a margin of the given value.
+func CheckAverageApproximateTo(num, err float64) Checker {
+	return Average(func(v float64) bool {
+		if math.Abs(num - v) > err {
+			return false
+		}
+		return true
```

Comment:
err -> margin (err should be used only for errors) and
`return mat.Abs(num - v) <= margin`